### ISSUE TITLE
Feature request: add a flag which run migrations with CREATE TABLE UNLOGGED instead of CREATE TABLE

### ISSUE THREAD

**Issue by capaj** [prisma/prisma#15381]
## Problem

integration/functional tests involving the DB are slow. One way to speed them up significantly is to use UNLOGGED tables

## Suggested solution

I am not sure how everyone else does it, but I setup DBs for testing using `prisma migrate reset` command.
So for me, the ideal solution would be a CLI flag for that command like `--unlogged-tables`.
This would do a very simple thing-when any `sql` files is loaded, before it is executed, we would just replace all `CREATE TABLE|create table` with `CREATE TABLE UNLOGGED` 

## Alternatives

alternatively I could write a script that replaces it in the migration files, calls  `prisma migrate reset`  and then reverts all migration files back, but the CLI flag would be much more elegant.

## Additional context

I write JS/TS for food and shelter.

**Comment by DavidHancu** [comment]
👋 **Hi @capaj!**

I've been working on my NPM Package [Prisma Util](https://www.npmjs.com/package/prisma-util), which is all about implementing features that are requested by developers in the community. This issue has attracted my attention and I want to ask if you'd be interested in it being implemented in Prisma Util.

Could you explain this feature a bit more so I can understand exactly what capabilities you want?

**Comment by marekhanzlik** [comment]
I think it would be benedficial to be able to specify `UNLOGGED` in the schema definition too. I use unlogged tables for cache (instead of redis to reduce TCO). Or is there another way to specify that in the model without modifying generated migrations?

**Comment by ahonn** [comment]
Is anyone working on this? I use unlogged tables for cache too, but Prisma looks like not support `CREATE TABLE UNLOGGED` yet.

**Comment by janpio** [comment]
You can always modify the generated `migration.sql` and the contained SQL statements before executing them. This issue is just about having a flag that allows to configure that, so the generated SQL is like that by default.

**Comment by ahonn** [comment]
@janpio 
Yes, I know that I can modify the generated `migration.sql` to add `UNLOGGED`. 
after I modify `migration.sql` and run `prisma migrate` next time, always has a `modified after it was applied` warning message. So I have to update the `checksum` of

**Comment by ahonn** [comment]
@janpio 
Yes, I know that I can add `UNLOGGED` manually to `migration.sql`, but it generates a warning message when I run `prisma migrate` after any modifications. To resolve it, I must update the `checksum` of the `_prisma_migrations` table. It would be easier if `prisma.schema` had a flag to handle this.

**Comment by janpio** [comment]
That should not be the case if you use `migrate dev --create-only` to create the migration file, adapt it, and only _then_ apply it (which will mark it as applied in `_prisma_migrations` with a checksum) with another `migrate dev` or `migrate deploy` run.
